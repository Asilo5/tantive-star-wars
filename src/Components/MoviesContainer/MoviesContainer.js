import React from 'react';
import MovieCard from '../MovieCard/MovieCard';
import './MoviesContainer.scss';
import NavBar from '../NavBar/NavBar';

const MoviesContainer = ({movies, userName, userQuote, userRank}) => {
    let allMovies = movies.map(movie => {
    return <MovieCard 
            title={movie.title}
            episode_id={movie.episode_id}
            release_date={movie.release_date}
            id={movie.episode_id}
            key={movie.episode_id}
            />
        }); 
    return (
        <section className='movie-section'>
            <NavBar serName={userName} userQuote={userQuote} userRank={userRank}/> 
            <section className='movie-container'>
              {allMovies}
            </section>
        </section>
    )
}

export default MoviesContainer;