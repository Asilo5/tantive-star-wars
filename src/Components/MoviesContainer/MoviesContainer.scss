.movie-section {
    display: flex;
}

.movie-container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
}