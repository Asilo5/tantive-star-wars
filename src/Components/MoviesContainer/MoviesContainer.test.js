import React from 'react';
import { shallow } from 'enzyme';
import MoviesContainer from './MoviesContainer';

describe('MoviesContainer', () => {

    it('should match snapshot with all information passing in correctly', () => {
        const wrapper = shallow(<MoviesContainer movies={[
            {
              title: 'The Phantom Menace',
              episode_id: 1,
              release_date: 2005,
              id: 1,
            },
            {
              title: 'Revenge of the Sith',
              episode_id: 3,
              release_date: 1977,
              id: 3,
            }
          ]} 
            getCharacters={jest.fn()} />
    
        )
        expect(wrapper).toMatchSnapshot();
    });

});