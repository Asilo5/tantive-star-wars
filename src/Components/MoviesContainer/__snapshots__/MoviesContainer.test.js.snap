// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviesContainer should match snapshot with all information passing in correctly 1`] = `
<section
  className="movie-section"
>
  <NavBar />
  <section
    className="movie-container"
  >
    <MovieCard
      episode_id={1}
      id={1}
      key="1"
      release_date={2005}
      title="The Phantom Menace"
    />
    <MovieCard
      episode_id={3}
      id={3}
      key="3"
      release_date={1977}
      title="Revenge of the Sith"
    />
  </section>
</section>
`;
